unittests/windowstests.py

```diff
@@ -398,3 +398,44 @@ def test_vsenv_option(self):
         with mock.patch.object(self, 'install_command', self.meson_command + ['install']):
             out = self.install(override_envvars=env)
             self.assertIn('Activating VS', out)
+
+    @skip_if_not_language('nasm')
+    def test_nasm_linker_args(self):
+        # TODO: there should be some way to query how we're linking things
+        # without resorting to reading the ninja.build file
+        if self.backend is not Backend.ninja:
+            raise SkipTest('This test reads the ninja file')
+
+        testdir = os.path.join(self.nasm_test_dir, '3 nasm only')
+        self.init(testdir)
+
+        env = get_fake_env(testdir, self.builddir, self.prefix)
+        cc = detect_c_compiler(env, MachineChoice.HOST)
+
+        build_ninja = os.path.join(self.builddir, 'build.ninja')
+        with open(build_ninja, encoding='utf-8') as f:
+            contents = f.read()
+
+        self.assertRegex(contents, r'build dummy(\.dll)?.*: nasm_LINKER')
+        if cc.get_id() == 'msvc':
+            # The following test DLL defaultlibs
+            # TODO: add an example for compiling a Windows executable from NASM
+            self.assertRegex(contents, r'LINK_ARGS =(.+)"/DEFAULTLIB:ucrt(d)?\.lib"')
+            self.assertRegex(contents, r'LINK_ARGS =(.+)".*/DEFAULTLIB:vcruntime(d)?\.lib"')
+            self.assertRegex(contents, r'LINK_ARGS =(.+)"/DEFAULTLIB:msvcrt(d)?\.lib"')
+
+    @skip_if_not_language('nasm')
+    def test_nasm_with_c_use_c_linker(self):
+        # TODO: there should be some way to query how we're linking things
+        # without resorting to reading the ninja.build file
+        if self.backend is not Backend.ninja:
+            raise SkipTest('This test reads the ninja file')
+
+        testdir = os.path.join(self.nasm_test_dir, '4 nasm and c together')
+        self.init(testdir)
+
+        build_ninja = os.path.join(self.builddir, 'build.ninja')
+        with open(build_ninja, encoding='utf-8') as f:
+            contents = f.read()
+
+        self.assertRegex(contents, r'build hello(\.exe)?.*: c_LINKER')
```

Comment:
Wondering if we could instead add `default_options: 'c_link_args=something'` and somehow make code that only works if we get that linker argument. That would be more a functional test rather than this unit test poking at internals. But I'm not sure how to do this exactly.